Add documentation tree.

erykoff opened the pull request:
This also adds plotting circles and Tissot Indicatrices.

damonge reviewed (commented):
Woah, super cool! The following feedback is mostly based on me browsing through the docs rather than the code,
- I found the docs super detailed and useful. Made me want to try the code out straight away. Good job!
- It would be good if the notebooks could show the results of running each cell.
- When reading the bit about drawing the milky way I wondered: "is skyproj meant to plot only maps in equatorial coords?". Might be worth clarifying this and perhaps document how one can change coordinates.
- The API docs for some parts of the code are still missing, although I think all or most of the important classes are there. When documenting those I think you're writing the docs right after `__init__`, whereas I thought the standard was to write them in between the `class` declaration and `__init__`.

erykoff commented:
@damonge Thank you so much for your feedback!  I think I've addressed all of them, and added more features which is maybe beyond what should be done for a "docs" ticket but 🤷 .
